core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -972,10 +987,17 @@ public Builder removeSnapshots(List<Snapshot> snapshotsToRemove) {
       }
 
       this.snapshots = retainedSnapshots;
-      if (!snapshotsById.containsKey(currentSnapshotId)) {
-        setCurrentSnapshot(null, System.currentTimeMillis());
+
+      // remove any refs that are no longer valid
```

Comment:
Should this fail if there are still refs?

Reply:
Based on the current usage, we are only calling this when expiring snapshots. At that time the snapshots send in here should already be the ones that have to be removed after evaluating all retention policies. I think that's why we directly remove the refs instead of failing.

Reply:
+1 I think it is in the scope for removeSnapshots(snapshots) to remove the references itself for any snapshots we are removing. Otherwise we would have to have the caller do a separate removeReferences and then do a removeSnapshots which seems more cumbersome.